scripts/gha/test_simulator.py

```diff
@@ -434,18 +434,35 @@ def _run_apple_gameloop_test(bundle_id, app_path, gameloop_app, device_id, retry
   return log
   
 
-def _install_apple_app(app_path, device_id):
+def _install_apple_app(app_path, device_id, retry=1):
   """Install integration_test app into the simulator."""
   args = ["xcrun", "simctl", "install", device_id, app_path]
   logging.info("Install testapp: %s", " ".join(args))
-  subprocess.run(args=args, check=True)
+  check = (retry <= 1)
+  result = subprocess.run(args=args, check=check)
+  if result.returncode != 0:
+    logging.info("Install testapp %s failed, Reset Simualtor now... Remaining retry: %s", app_path, retry-1)
+    args = ["xcrun", "simctl", "erase", device_id]
+    logging.info("Erase my simulator: %s", " ".join(args))
+    subprocess.run(args=args, check=True)
+    args = ["xcrun", "simctl", "boot", device_id]
+    logging.info("Reboot my simulator: %s", " ".join(args))
+    subprocess.run(args=args, check=True)
+    _install_apple_app(app_path, device_id, retry=retry-1)
 
 
-def _uninstall_apple_app(bundle_id, device_id):
+def _uninstall_apple_app(bundle_id, device_id, retry=1):
   """Uninstall integration_test app from the simulator."""
   args = ["xcrun", "simctl", "uninstall", device_id, bundle_id]
   logging.info("Uninstall testapp: %s", " ".join(args))
-  subprocess.run(args=args, check=True)
+  check = (retry <= 1)
+  result = subprocess.run(args=args, check=check)
+  if result.returncode != 0:
+    logging.info("Uninstall testapp %s failed, Reset Simualtor now... Remaining retry: %s", bundle_id, retry-1)
+    args = ["xcrun", "simctl", "boot", device_id]
+    logging.info("Reboot my simulator: %s", " ".join(args))
+    subprocess.run(args=args, check=True)
+    _uninstall_apple_app(bundle_id, device_id, retry=retry-1)
```

Comment:
Is it worth adding a short delay, just in case there's a transient error that might be resolved in a few moments?

Also, WDYT about adding a general "run_with_retry" function?

Reply:
According to the [post](https://stackoverflow.com/questions/40327935/failed-to-load-info-plist-from-bundle), it doesn't look like a transient error. 

I was thinking about run_with_retry. However, here, we are not only retry the cmd, but also reset and reboot the simulator.